module/zfs/spa.c

```diff
@@ -1420,19 +1420,99 @@ spa_config_parse(spa_t *spa, vdev_t **vdp, nvlist_t *nv, vdev_t *parent,
 	return (0);
 }
 
+static boolean_t
+spa_should_flush_logs_on_unload(spa_t *spa)
+{
+	if (!spa_feature_is_active(spa, SPA_FEATURE_LOG_SPACEMAP))
+		return (B_FALSE);
+
+	if (!spa_writeable(spa))
+		return (B_FALSE);
+
+	if (!spa->spa_sync_on)
+		return (B_FALSE);
+
+	if (spa_state(spa) != POOL_STATE_EXPORTED)
+		return (B_FALSE);
+
+	if (zfs_keep_log_spacemaps_at_export)
+		return (B_FALSE);
+
+#ifdef ZFS_DEBUG
+	/*
+	 * At this point we always want to flush as many metaslabs
+	 * as we can. That said though, for debug builds we try to
+	 * do this half the time, and not flush the other half
+	 * so the log space map loading paths in the next pool
+	 * import get exercised too.
+	 */
+	return (spa_get_random(2) == 0);
```

Comment:
I can take the non-deterministic step out altogether since this is now a stable feature and if we really want to keep the spacemaps at export, there exists the `zfs_keep_log_spacemaps_at_export` tunable also used by a test in the test-suite.

Reply:
Removed